### PR TITLE
cherry-pick: "Propagate WireguardHostEncryptionEnabled config_param (#2897)"

### PR DESCRIPTION
**Pull request by aaaaaaaalex**
This PR picks commit `bd0257b3abed4828f71387ef75156688c658b314` from felix/master

### CODE REVIEW

**Comment by aaaaaaaalex** [comment]
Closing and waiting till post-release to get this work into the relevant streams!

**Comment by aaaaaaaalex** [comment]
@caseydavenport I'm happy to merge this whenever, if you're around to give approval :)

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**
